synapse/handlers/register.py

```diff
@@ -812,7 +812,10 @@ class and RegisterDeviceReplicationServlet.
             assert access_token_expiry is None
             access_token = self.macaroon_gen.generate_guest_access_token(user_id)
         else:
-            if should_issue_refresh_token:
+            if (
+                should_issue_refresh_token
+                and self.refreshable_access_token_lifetime is not None
+            ):
```

Comment:
Bugfix?

Even if so, it's in between unreleased versions so probably doesn't need calling out in the changelog.

Reply:
(I wonder if `should_issue_refresh_token and self.refreshable_access_token_lifetime is None` should raise an error?)

Reply:
It's just because Mypy can't realise that `should_issue_refresh_token` is only true if there is a lifetime configured (it's set several levels up).

Maybe `assert self.refreshable_access_token_lifetime is not None` makes more sense, though I'm not entirely sure.

Reply:
Putting it in the `if` condition makes me think that the condition must be considered at runtime. My preference would be an assertion with a comment like 

> Mypy can't realise that should_issue_refresh_token is only true if there is a lifetime configured (it's set several levels up).